pocs/sensors/sensor_recorder.py

```diff
@@ -0,0 +1,111 @@
+import queue as queue_module
+import threading
+
+
+class SensorRecorder(threading.Thread):
+    """Monitors a queue for readings from sensors, records in the store.
+
+    This is a Thread sub-class. This implies that it can only be started
+    running once (threading.Thread does not allow restarting).
+
+    This is the format in which we expect entries in the queue:
+        {'name': board_name, 'timestamp': timestamp_string, 'data': reading}
+    """
+
+    def __init__(self,
+                 save_func,
+                 send_func,
+                 logger,
```

Comment:
Let's do that in another CL. We could build on that FakeLogger to create a "stdout" logger.

Reply:
This make this whole module different from everything that is in the `pocs` folder except for what is in `utils` (and this feels like a util). Then again, this does run differently and is controlled by something other than a POCS instance, which is why the original PEAS separation existed. It doesn't feel consistent with the rest of the setup.

Reply:
I agree. But, what is going to happen to that consistent code as you add more dependency injection? And we have the problem that PanBase assumes POCS and not PEAS.

FWIW, I've been wondering whether we could use some kind of registry class, supporting dependency injection. This would be analogous to the config dictionary, but for classes rather than values. Of course, I'm sure we can find plenty of frameworks with such features.